src/pybind/mgr/cephadm/module.py

```diff
@@ -1037,8 +1034,65 @@ def create_osds(self, drive_group):
         return self.get_hosts().then(lambda hosts: self._create_osd(hosts, drive_group))
 
     @with_services('osd')
-    def remove_osds(self, osd_ids, services):
-        # type: (List[str], List[orchestrator.ServiceDescription]) -> AsyncCompletion
+    def remove_osds(self, osd_ids, destroy=False, services=[]):
+        # type: (List[str], bool, List[orchestrator.ServiceDescription]) -> AsyncCompletion
+
+        def ok_to_destroy():
+            return self.mon_command({
+                'prefx': 'osd safe-to-destroy',
+                'ids': osd_ids,
+            })
+        if destroy:
```

Comment:
`destroy`ing an OSD requires an OSD to be down, hence either a failed or a stopped systemd unit. Therefore this call can't be as straightforward unfortunately.